src/pretix/presale/views/customer.py

```diff
@@ -541,3 +568,286 @@ def get(self, request, *args, **kwargs):
+        redirect_uri = build_absolute_uri(self.request.organizer, 'presale:organizer.customer.login.return', kwargs={
+            'provider': self.provider.pk
+        })
+
+        if self.provider.method == "oidc":
+            return redirect(oidc_authorize_url(self.provider, f'{nonce}#{next_url}', redirect_uri))
+        else:
+            raise Http404("Unknown SSO method.")
+
+    def get_success_url(self):
+        url = self.get_redirect_url()
+
+        if not url:
+            return eventreverse(self.request.organizer, 'presale:organizer.customer.profile', kwargs={})
+        return url
+
+
+class SSOLoginReturnView(RedirectBackMixin, View):
+    """
+    Start logging in with a SSO provider.
+    """
+    form_class = AuthenticationForm
+    redirect_authenticated_user = True
+
+    @method_decorator(sensitive_post_parameters())
+    @method_decorator(csrf_protect)
+    @method_decorator(never_cache)
+    def dispatch(self, request, *args, **kwargs):
+        if not request.organizer.settings.customer_accounts:
+            raise Http404('Feature not enabled')
+        if self.redirect_authenticated_user and self.request.customer:
+            redirect_to = self.get_success_url()
+            if redirect_to == self.request.path:
+                raise ValueError(
+                    "Redirection loop for authenticated user detected. Check that "
+                    "your LOGIN_REDIRECT_URL doesn't point to a login page."
+                )
+            return HttpResponseRedirect(redirect_to)
+        r = super().dispatch(request, *args, **kwargs)
+        request.session.pop(f'pretix_customerauth_{self.provider.pk}_nonce', None)
+        request.session.pop(f'pretix_customerauth_{self.provider.pk}_popup_origin', None)
+        request.session.pop(f'pretix_customerauth_{self.provider.pk}_cross_domain_requested', None)
+        return r
+
+    @cached_property
+    def provider(self):
+        return get_object_or_404(self.request.organizer.sso_providers.filter(is_active=True), pk=self.kwargs['provider'])
+
+    def get(self, request, *args, **kwargs):
+        redirect_to = None
+        popup_origin = None
+
+        if request.session.get(f'pretix_customerauth_{self.provider.pk}_popup_origin'):
+            popup_origin = request.session[f'pretix_customerauth_{self.provider.pk}_popup_origin']
+
+        if self.provider.method == "oidc":
+            if not request.GET.get('state'):
+                return self._fail(
+                    _('Login was not successful. Error message: "{error}".').format(
+                        error='state parameter missing',
+                    ),
+                    popup_origin,
+                )
+
+            nonce, redirect_to = request.GET['state'].split('#')
+
+            if nonce != request.session[f'pretix_customerauth_{self.provider.pk}_nonce']:
```

Comment:
I stumbled into this error, because of redirect error in my setup. So it should not happen IRL, but if something is wrong with the serssion whatsoever, this should not throw a 500 IMHO. So I suggest doing:

```suggestion
            if nonce != request.session.get(f'pretix_customerauth_{self.provider.pk}_nonce'):
```